cdc/capture/capture.go

```diff
@@ -542,27 +569,73 @@ func (c *Capture) AsyncClose() {
 }
 
 // Drain removes tables in the current TiCDC instance.
-func (c *Capture) Drain() <-chan struct{} {
-	c.liveness.Store(model.LivenessCaptureStopping)
-	log.Info("draining the capture, removing all tables in the capture")
+func (c *captureImpl) Drain(ctx context.Context) <-chan struct{} {
+	log.Info("draining capture, removing all tables in the capture")
 
-	// Disable drain for now.
-	//
-	// TODO close done when
-	// 1. there is no table in the capture, and
-	// 2. the capture is not owner.
+	const drainInterval = 100 * time.Millisecond
 	done := make(chan struct{})
-	close(done)
+	go func() {
+		defer close(done)
+		ticker := time.NewTicker(drainInterval)
+		defer ticker.Stop()
+		for {
```

Comment:
how about rewrite the for loop like

```
		for {
			select {
			case <- ticker.C:
				if c.drainImpl(ctx) {
					return
				}
			case <- ctx.Done():
				return
			}
		}
```

Reply:
> how about rewrite the for loop like
> 
> ```
> 		for {
> 			select {
> 			case <- ticker.C:
> 				if c.drainImpl(ctx) {
> 					return
> 				}
> 			case <- ctx.Done():
> 				return
> 			}
> 		}
> ```

It waits a tick before the first drain, relevantly slow.